core/shared/utils/bh_atomic.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2023 Amazon Inc.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef _BH_ATOMIC_H
+#define _BH_ATOMIC_H
+
+#include "gnuc.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef uint32 bh_atomic_32_t;
+
+#if defined(__GNUC_PREREQ)
+#if __GNUC_PREREQ(4, 7)
+#define CLANG_GCC_HAS_ATOMIC_BUILTIN
+#endif
+#elif defined(__clang__)
+#if __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 0)
+#define CLANG_GCC_HAS_ATOMIC_BUILTIN
+#endif
+#endif
+
+#if defined(CLANG_GCC_HAS_ATOMIC_BUILTIN)
+#define BH_ATOMIC_32_IS_ATOMIC 1
```

Comment:
Using `BH_ATOMIC_32_IS_ATOMIC` is a little confusing, how about renaming it to `BH_UINT32_IS_ATOMIC`?

Reply:
PS: It is good if it refers to type `bh_atomic_32_t`.

Reply:
`BH_ATOMIC_32_IS_ATOMIC` means if `bh_atomic_32_t` is really atomic.
that is, if it can be accessed atomically with macros like `BH_ATOMIC_32_FETCH_OR` without extra lock.

Reply:
OK, got it.